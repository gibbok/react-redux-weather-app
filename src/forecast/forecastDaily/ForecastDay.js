import React from 'react'
import moment from 'moment'
import IconWeather from '../../shared/icon/IconWeather'

const ForecastDay = ({ onClick, date, tempMin, tempMax, iconCode, weatherDescription }) => {
  const dateFormat = moment.unix(date).format('ddd, MMM D')
  const tempMinRounded = Math.round(tempMin)
  const tempMaxRounded = Math.round(tempMax)
  return (
    <div onClick={onClick}>
      <div>{dateFormat}</div>
      <div>
        <IconWeather code={iconCode} />
      </div>
      <div>
        <div>
          {tempMinRounded}&#176;
        </div>
        <div>
          {tempMaxRounded}&#176;
        </div>
      </div>
      <div>
        {weatherDescription}
      </div>
    </div>
  )
}
export default ForecastDay
