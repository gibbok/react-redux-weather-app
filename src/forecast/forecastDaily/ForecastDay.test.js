import React from 'react'
import {shallow} from 'enzyme'
import ForecastDay from './ForecastDay'
import toJson from 'enzyme-to-json'

describe('<ForecastDay />', () => {
  it('should render', () => {
    const wrapper = shallow(
      <ForecastDay
        onClick={() => {}}
        date={1504811633838}
        tempMin={10}
        tempMax={15}
        iconCode={200}
        weatherDescription={'storm showers'}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
