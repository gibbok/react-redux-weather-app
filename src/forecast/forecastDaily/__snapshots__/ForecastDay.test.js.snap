// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastDay /> should render 1`] = `
<div
  onClick={[Function]}
>
  <div>
    Sun, Aug 1
  </div>
  <div>
    <IconWeather
      code={200}
    />
  </div>
  <div>
    <div>
      10
      °
    </div>
    <div>
      15
      °
    </div>
  </div>
  <div>
    storm showers
  </div>
</div>
`;
